File: vol2/ch15/LeetCode1863.cpp
class Solution {
public:
    int subsetXORSum(vector<int>& nums) {
        int n=nums.size();
        int ans=0;
        for(int V=0;V<=(1<<n)-1;V++) {
            int curs=0;
            for(int j=0;j<n;j++) {
                if(inset(V,j)) curs^=nums[j];
            }
            ans+=curs;
        }
        return ans;
    }
    bool inset(int V,int j) {       //判断下标j是否在V中
        return (V & (1<<j))!=0;
    }
};

File: vol2/ch15/LeetCode1863.py
class Solution:
    def subsetXORSum(self, nums: List[int]) -> int:
        n,ans=len(nums),0
        for V in range(0,1<<n):
            curs=0
            for j in range(0,n):
                if self.inset(V,j):curs^=nums[j]
            ans+=curs
        return ans

    def inset(self,V,j):    #判断下标j是否在V中
        return (V & (1<<j))!=0
